src/t8_schemes/t8_default/t8_default_pyramid/t8_default_pyramid_cxx.hxx

```diff
@@ -44,293 +44,573 @@ public:
+   * \param [in,out] dest This element's entries will be overwritted with the
+   *                    entries of \b source.
+   * \note \a source and \a dest may point to the same element.
+   */
   virtual void        t8_element_copy (const t8_element_t *source,
                                        t8_element_t *dest);
 
-/** Given a boundary face inside a root tree's face construct
- *  the element inside the root tree that has the given face as a
- *  face */
-  virtual int         t8_element_extrude_face (const t8_element_t *face,
-                                               const t8_eclass_scheme_c
-                                               *face_scheme,
-                                               t8_element_t *elem,
-                                               int root_face);
-
-/** Compare two elements. returns negative if elem1 < elem2, zero if elem1 equals elem2
- *  and positiv if elem1 > elem2.
- *  If elem2 is a copy of elem1 then the elements are equal.
- */
+  /** Compare two elements.
+    * \param [in] elem1  The first element.
+    * \param [in] elem2  The second element.
+    * \return       negativ if elem1 < elem2, zero if elem1 equals elem2
+    *               and positiv if elem1 > elem2.
+    *  If elem2 is a copy of elem1 then the elements are equal.
+    */
   virtual int         t8_element_compare (const t8_element_t *elem1,
                                           const t8_element_t *elem2);
 
-/** Construct the parent of a given element. */
+  /** Compute the parent of a given element \b elem and store it in \b parent.
+   *  \b parent needs to be an existing element. No memory is allocated by this function.
+   *  \b elem and \b parent can point to the same element, then the entries of
+   *  \b elem are overwritten by the ones of its parent.
+   * \param [in] elem   The element whose parent will be computed.
+   * \param [in,out] parent This element's entries will be overwritten by those
+   *                    of \b elem's parent.
+   *                    The storage for this element must exist
+   *                    and match the element class of the parent.
+   *                    For a pyramid, for example, it may be either a
+   *                    tetrahedron or a pyramid depending on \b elem's childid.
+   */
   virtual void        t8_element_parent (const t8_element_t *elem,
                                          t8_element_t *parent);
 
-/** Construct a same-size sibling of a given element. */
+  /** Compute the number of siblings of an element. That is the number of 
+   * Children of its parent.
+   * \param [in] elem The element.
+   * \return          The number of siblings of \a element.
+   * Note that this number is >= 1, since we count the element itself as a sibling.
+   */
+  virtual int         t8_element_num_siblings (const t8_element_t *elem)
+    const;
+
+  /** Compute the number of siblings of an element. That is the number of 
```

Comment:
```suggestion
  /** Compute a specific sibling of \a elem
```

Reply:
I am not sure that I understand this suggestion

Reply:
I have understood it now